### PR TITLE
Move the jump tables into the source file to reduce number of seek() calls by half.

### PR DESCRIPTION
**Pull request by dalf**
Partial implementation of #4

The jump_table is embedded inside ```__main__.py```, perhaps it should be another file.

[EDIT] an additional way to minimize disk access is to store the small layers directly into the _JUMPTABLE . The layers are base64 encoded to avoid NSFW words in the source code. See https://gist.github.com/dalf/8e16440d59401e95556d68f3d5b07c14
It removes disk access for '.com', '.net', etc, so most of the time there is only one disk access.

I haven't analyze the memory footprints of the two versions.

### CODE REVIEW

File: hstspreload/__init__.py [hunk 1]
```diff
@@ -2,13 +2,16 @@
...
..., (None, 0), (None, 0), (None, 0), (None, 0), (None, 0), (None, 0), (None, 0), (None, 0), (1785802, 42), (1785844, 21), (1785865, 21), (None, 0), (1785886, 42), (1785928, 25), (None, 0), (None, 0), (1785953, 21), (1785974, 42), (None, 0), (None, 0), (1786016, 21), (1786037, 19), (1786056, 26), (None, 0), (None, 0), (None, 0), (1786082, 21), (None, 0), (None, 0), (1786103, 38), (None, 0), (1786141, 22), (1786163, 21), (1786184, 21), (None, 0), (None, 0), (1786205, 63), (None, 0), (1786268, 21), (1786289, 42), (None, 0), (1786331, 17), (None, 0), (None, 0), (None, 0), (None, 0), (1786348, 21), (1786369, 21), (None, 0), (None, 0), (1786390, 21), (None, 0), (None, 0), (1786411, 21), (None, 0), (1786432, 26), (None, 0), (1786458, 50), (None, 0), (None, 0), (None, 0), (1786508, 50), (1786558, 26), (1786584, 21), (1786605, 21), (1786626, 19), (None, 0), (1786645, 35), (1786680, 26), (1786706, 23), (1786729, 21), (1786750, 42), (None, 0), (None, 0), (None, 0), (None, 0), (None, 0), (None, 0), (1786792, 21), (None, 0), (None, 0), (None, 0), (1786813, 21), (None, 0), (None, 0), (1786834, 90), (None, 0), (1786924, 239), (1787163, 38), (None, 0), (None, 0), (None, 0), (None, 0)]]  # noqa: E501
```
[the remark this answers is not in the document]
Comment (dalf) [reply hstspreload/__init__.py]: Yes clearly.

File: hstspreload/__init__.py [hunk 2]
```diff
@@ -2,13 +2,16 @@
...
+# fmt: on
```
[the remark this answers is not in the document]
Comment (dalf) [reply hstspreload/__init__.py]: Yes. Done in commit 26a10826a90a3a983206edde6c43ef470520b6d3

File: hstspreload/__init__.py [hunk 3]
```diff
@@ -103,12 +105,12 @@ def in_hsts_preload(host: typing.AnyStr) -> bool:
                 return False
 
             # Read the jump table for the layer and label
-            offset, size = _get_offset_and_size(f, layer, label)
-            if offset == 0:
+            offset, size = _JUMPTABLE[layer][_crc8(label)]
```
[the remark this answers is not in the document]
Comment (dalf) [reply hstspreload/__init__.py]: Done in commit 26a10826a90a3a983206edde6c43ef470520b6d3

**Comment by dalf** [comment]
About import speed, I'm not sure how to make a reliable benchmark.

A micro benchmark on my laptop (i7-8550U CPU, Python 3.8, Ubuntu 20.04) : 

```sh
#!/bin/bash
for i in {0..100}
do 
    # rm hstspreload/__pycache__/__init__.cpython-38.pyc
    python test_import.py
done
```

```python
import time
before = time.time()
import hstspreload
after = time.time()
print(after-before)
```I guess the kernel caches all / most on the files.



The results:

|         | master, existing .pyc | jumptable, existing .pyc | master, no .pyc | jumptable, no .pyc |
|---------|-----------------------|--------------------------|-----------------|--------------------|
| Average | 0,0092                | 0,0091                   | 0,0116          | 0,0158             |
| Median  | 0,0090                | 0,0089                   | 0,0114          | 0,0155             |

It seems the import time is nearly identical, but I guess the kernel caches all / most on the files.

File: hstspreload/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dalf) [reply hstspreload/__init__.py]: flake8 ignores the line, but black is not happy without the ```#fmt: on|off```.

**Comment by sethmlarson** [comment]
I tested this in a resource-constrained Docker container (256MiB) and saw similar results for import times of your branch and master. I made one commit to your branch removing the intermediate # fmt: on/off, seemed to pass lint locally for me.

This PR also makes me realize we should automate the `_GTLD_INCLUDE_SUBDOMAINS` value as well to ensure that the fast-path is automatically added for all gTLDs that may be added in the future.

**Pull request merged**